src/System.Runtime.Extensions/tests/VoidMainWithExitCodeApp/VoidMainWithExitCodeApp.cs

```diff
@@ -4,16 +4,39 @@
 
 using System;
 using System.Reflection;
+using System.Threading;
 
 namespace VoidMainWithExitCodeApp
 {
     internal static class Program
     {
         static void Main(string[] args)
         {
-            int exitCode = args.Length > 0 ? int.Parse(args[0]) : 0;
-            typeof(Environment).GetTypeInfo().GetDeclaredProperty("ExitCode").SetValue(null, exitCode);
-            // Environment.ExitCode = exitCode; // TODO: Remove reflection when package updated with latest Environment exposing ExitCode
+            int exitCode = int.Parse(args[0]);
+            int mode = int.Parse(args[1]);
+
+            PropertyInfo set_ExitCode = typeof(Environment).GetTypeInfo().GetDeclaredProperty("ExitCode");
+            MethodInfo Exit = typeof(Environment).GetTypeInfo().GetDeclaredMethod("Exit");
+
+            switch (mode)
+            {
+                case 1: // set ExitCode and exit
+                    set_ExitCode.SetValue(null, exitCode); // TODO: Environment.ExitCode = exitCode;
+                    break;
+
+                case 2: // set ExitCode, exit, and then set ExitCode from another foreground thread
+                    new Thread(() => // foreground thread
+                    {
+                        Thread.Sleep(1000); // time for Main to exit
```

Comment:
Not really.  The goal here is to get the thread to set the property after Main has already exited; to set an event, it'd be the Main method setting it, which would significantly increase the likelihood the property got set before Main actually returned.  We could set the event from somewhere else, like a handler for the AssemblyLoadContext.Default.Unloading event, and that might be reasonable to do in the future (plus I do separately want to add a test that sets the property directly from that), but that type isn't available to be used on desktop and these tests are configured to run on both core and desktop, so I need to spend a bit of time first getting things separated so only some tests are compiled/configured to run on both while others are compiled/configured to run only on core.